operatorPointwise_unary_test.go

```diff
@@ -430,12 +430,12 @@ func TestTanhDiff(t *testing.T) {
 		t.Error(err)
 	}
 
-	correct := 1.0 - (math.Tanh(v) * math.Tanh(v)) // I'm surprised Golang doesn't have a secant function!
-	assert.Equal(correct, x.boundTo.(*dualValue).d.Data())
+	correct := 1.0 - (float64(math.Tanh(v)) * float64(math.Tanh(v))) // I'm surprised Golang doesn't have a secant function!
+	assert.InDeltaf(correct, x.boundTo.(*dualValue).d.Data(), 1e-14, "")
 
 	// Tensor edition
 	xdvd := xT.boundTo.(*dualValue).d.(*tensor.Dense)
-	assert.Equal([]float64{correct, correct}, xdvd.Data())
+	assert.InDeltaSlicef([]float64{correct, correct}, xdvd.Data(), 1e-14, "")
```

Comment:
This is good enough for the test.
However, if we don't find a proper fix(that validates the equal asertions), we may eventually face another issue that is related. WDYT?

Reply:
I think a difference of 1e-14 should not significant, but let's see what @chewxy thinks about this

Reply:
Can you add a comment on why this is used (i.e. no guarantees of identical behaviours across architecture) - link to https://github.com/golang/go/issues/44528#issuecomment-783975492

Reply:
done